src/alerts/check_signals_updates.R

```diff
@@ -0,0 +1,193 @@
+box::use(jsonlite)
+box::use(httr[GET, POST, add_headers, status_code, content])
+box::use(httr2)
+box::use(purrr)
+box::use(logger)
+box::use(stringr)
+
+box::use(../utils/get_env[get_env])
+box::use(../utils/hs_logger)
+box::use(cs = ../utils/cloud_storage)
+
+hs_logger$configure_logger()
+
+#' Builds and posts a message to Slack, using incoming webhooks
+#' See API docs: https://api.slack.com/messaging/webhooks
+#'
+#' @param header_text Text for the message header
+#' @param status_text Text for the GitHub actions status report
+#' @param signals_text Text for the signals status report
+#'
+#' @returns Nothing. Message is posted to slack and will log an error if not successful
+slack_post_message <- function(header_text, status_text, signals_text) {
+  # See https://app.slack.com/block-kit-builder for prototyping layouts in JSON
+  msg <- list(
+    blocks = list(
+      list(
+        type = "section",
+        text = list(
+          type = "mrkdwn",
+          text = header_text
+        )
+      ),
+      list(
+        type = "section",
+        text = list(
+          type = "mrkdwn",
+          text = signals_text
+        )
+      ),
+      list(type = "divider"),
+      list(
+        type = "context",
+        elements = list(
+          list(
+            type = "mrkdwn",
+            text = status_text
+          )
+        )
+      ),
+      list(type = "divider")
+    )
+  )
+
+  # Create and perform the POST request using httr2
+  response <- httr2$request(get_env("HS_SLACK_URL")) |>
+    httr2$req_body_json(msg) |>
+    httr2$req_perform()
+
+  if (response$status_code != 200) {
+    logger$log_error("Error posting Slack message")
+    stop()
+  }
+}
+
+#' Builds the header text, depending on how many signals are reported
+#'
+#' @param n_signals Number of signals reported
+#'
+#' @returns String header text
+slack_build_header <- function(n_signals) {
+  if (n_signals == 0) {
+    paste0(Sys.Date(), ": No signals identified")
+  } else {
+    paste0(":rotating_light: <!channel> ", Sys.Date(), ": ", n_signals, " alerts identified")
+  }
+}
+
+#' Builds the signal alert text
+#'
+#' @param indicator_id Indicator ID
+#' @param df DataFrame with all the signals for the given indicator
+#'
+#' @returns String signal alert text
+slack_build_alert <- function(indicator_id, df) {
+  paste0(
+    "*",
+    indicator_id,
+    "* - ",
+    nrow(df),
+    " countries impacted - <",
+    df$campaign_url_archive[1],
+    " | See draft campaign>\n"
+  )
+}
+
+#' Takes the response from a GitHub Actions run of a single indicator
+#' and outputs a status message to be posted to Slack
+#'
+#' @param indicator_id ID of the indicator
+#'
+#' @returns String status message to be posted to Slack
+slack_build_workflow_status <- function(indicator_id) {
+  workflow_id <- paste0("monitor_", indicator_id, ".yaml")
+  base_logs_url <- "https://github.com/ocha-dap/hdx-signals/actions/runs/"
+
+  df_runs <- tryCatch({
```

Comment:
I'm not totally following how this would be implemented with the `tryCatch()` flow. I've funtionalized the GH API call, which I think simplifies things. What do you think of the readability now?